crates/router/src/db/refund.rs

```diff
@@ -670,12 +679,29 @@ impl RefundInterface for MockDb {
 
     async fn update_refund(
         &self,
-        _this: storage_types::Refund,
-        _refund: storage_types::RefundUpdate,
+        this: storage_types::Refund,
+        refund: storage_types::RefundUpdate,
         _storage_scheme: enums::MerchantStorageScheme,
     ) -> CustomResult<storage_types::Refund, errors::StorageError> {
-        // [#172]: Implement function for `MockDb`
-        Err(errors::StorageError::MockDbError)?
+        match self
+            .refunds
+            .lock()
+            .await
+            .iter_mut()
+            .find(|refund| this.refund_id == refund.refund_id)
+            .map(|r| {
+                let refund_updated = RefundUpdateInternal::from(refund).create_refund(r.clone());
+                *r = refund_updated.clone();
+                refund_updated
+            }) {
+            Some(refund_updated) => Ok(refund_updated),
+            None => {
+                return Err(errors::StorageError::ValueNotFound(
+                    "cannot find refund to update".to_string(),
+                )
+                .into())
+            }
```

Comment:
You can use [`ok_or_else()`](https://doc.rust-lang.org/std/option/enum.Option.html#method.ok_or_else) here.